view/src/storage.rs

```diff
@@ -343,37 +399,40 @@ impl Storage {
 
         let nullifier_bytes = nullifier.0.to_bytes().to_vec();
 
-        async move {
-            // Check if we already have the nullifier in the set of spent notes
-            if let Some(record) = sqlx::query!(
-                "SELECT nullifier, height_spent FROM spendable_notes WHERE nullifier = ?",
-                nullifier_bytes,
-            )
-            .fetch_optional(&pool)
-            .await?
-            {
-                let spent = record.height_spent.is_some();
+        // Check if we already have the nullifier in the set of spent notes
+        if let Some(height_spent) = spawn_blocking(move || {
+            pool.get()?
+                .prepare_cached("SELECT height_spent FROM spendable_notes WHERE nullifier = ?1")?
+                .query_and_then([nullifier_bytes], |row| {
```

Comment:
Maybe this would be a good place for some temporary `tracing::warn!` statements to what the query is returning?